packages/forms/src/styled/file-upload/StyledFileUpload.ts

```diff
@@ -0,0 +1,110 @@
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import styled, { css, ThemeProps, DefaultTheme } from 'styled-components';
+import { rgba } from 'polished';
+import { retrieveComponentStyles, DEFAULT_THEME, getColor } from '@zendeskgarden/react-theming';
+import { StyledLabel } from '../common/StyledLabel';
+import { StyledHint } from '../common/StyledHint';
+import { StyledMessage } from '../common/StyledMessage';
+
+const COMPONENT_ID = 'forms.file_upload';
+
+interface IStyledFileUploadProps {
+  isDragging?: boolean;
+  isCompact?: boolean;
+}
+
+const positionStyles = (props: ThemeProps<DefaultTheme> & IStyledFileUploadProps) => {
+  const topMargin = `${props.theme.space.base * (props.isCompact ? 1 : 2)}px`;
+
+  return css`
+    /* stylelint-disable */
+    ${StyledLabel} + &,
+    ${StyledHint} + &,
+    ${StyledMessage} + &,
+    & + ${StyledHint},
+    & + ${StyledMessage} {
+      margin-top: ${topMargin};
+    }
+    /* stylelint-enable */
+  `;
+};
+
+const colorStyles = (props: ThemeProps<DefaultTheme> & IStyledFileUploadProps) => {
+  const baseColor = getColor('primaryHue', 600, props.theme);
+  const hoverColor = getColor('primaryHue', 700, props.theme);
+  const activeColor = getColor('primaryHue', 800, props.theme);
+  const disabledBackgroundColor = getColor('primaryHue', 200, props.theme);
+  const disabledForegroundColor = getColor('primaryHue', 400, props.theme);
+  const boxShadow = `inset ${props.theme.shadows.md(rgba(baseColor as string, 0.35))}`;
+
+  return css`
+    border-color: ${props.isDragging ? activeColor : getColor('neutralHue', 600, props.theme)};
+    background-color: ${props.isDragging && rgba(baseColor as string, 0.2)};
+    color: ${props.isDragging ? activeColor : baseColor};
+
+    &:hover {
+      border-color: ${hoverColor};
+      background-color: ${rgba(baseColor as string, 0.08)};
+      color: ${hoverColor};
+    }
+
+    &[data-garden-focus-visible] {
+      box-shadow: ${boxShadow};
+    }
+
+    &:active {
+      border-color: ${activeColor};
+      background-color: ${rgba(baseColor as string, 0.2)};
+      color: ${activeColor};
+    }
+
+    &:disabled {
+      border-color: ${disabledForegroundColor};
+      background-color: ${disabledBackgroundColor};
+      color: ${disabledForegroundColor};
+    }
+  `;
+};
+
+export const StyledFileUpload = styled.div.attrs({
+  'data-garden-id': COMPONENT_ID,
+  'data-garden-version': PACKAGE_VERSION
+})<IStyledFileUploadProps>`
+  direction: ${props => (props.theme.rtl ? 'rtl' : 'ltr')};
+  /* prettier-ignore */
+  transition:
+    border-color 0.25s ease-in-out,
+    box-shadow 0.1s ease-in-out,
+    background-color 0.25s ease-in-out,
+    color 0.25s ease-in-out;
+  border: dashed ${props => props.theme.borderWidths.sm};
+  border-radius: ${props => props.theme.borderRadii.md};
+  cursor: pointer;
+  padding: ${props => `${props.theme.space.base * 5}px ${props.theme.space.base * 15}px`};
```

Comment:
At worst, the component should be given a `min-width` that is the sum of (border-right + padding-right + padding-left + border-right) in order to protect styling breakage at extreme widths.

@steue may want to comment about adding additional `em` space for the minimum amount of text to show in the center.

Moving all of this to a `sizeStyles` as [directed](https://github.com/zendeskgarden/react-components/blob/main/docs/api.md#rules) would have the added benefit of cleaning up the combinations & inline calculations.

Reply:
Do we provide a min-width for other block elements? I don't think `Input` includes one

Reply:
For other form elements? Yes. See dropdowns. We want to protect against broken layout styling seen here.

<img width="138" alt="Screen Shot 2020-09-03 at 6 14 20 PM" src="https://user-images.githubusercontent.com/143773/92188407-5a983280-ee11-11ea-8ad9-97865731e56d.png">

Even a small `min-width: 4em` to accommodate a longer word will help here.

Reply:
Please note that `box-sizing` will need to be set in order for the chosen `min-width` to play nice with CSS Bedrock.

Reply:
`4em` seems to work well with this combination. Updated.

![image](https://user-images.githubusercontent.com/4030377/92262931-b2c24980-ee90-11ea-8e83-46c220c52846.png)